store/postgres/src/relational/prune.rs

```diff
@@ -0,0 +1,331 @@
+use std::{fmt::Write, sync::Arc, time::Instant};
+
+use diesel::{
+    connection::SimpleConnection,
+    sql_query,
+    sql_types::{BigInt, Integer, Nullable},
+    Connection, PgConnection, RunQueryDsl,
+};
+use graph::{
+    components::store::PruneReporter,
+    prelude::{BlockNumber, CancelHandle, CancelToken, CancelableError, CheapClone, StoreError},
+    slog::Logger,
+};
+use itertools::Itertools;
+
+use crate::{
+    catalog,
+    copy::AdaptiveBatchSize,
+    deployment,
+    relational::{Table, VID_COLUMN},
+};
+
+use super::{Layout, SqlName};
+
+/// Utility to copy relevant data out of a source table and into a new
+/// destination table and replace the source table with the destination
+/// table
```

Comment:
Worth commenting somewhere on this file (maybe as a module-level comment) on the approach for the pruning implementation. In particular I'd naively expect it to be implemented by deleting rows, but I see we're copying to a new table first.

Reply:
Good point. I expanded the comment for `prune_by_copying` to explain that better since I envision that we might have a `prune_by_deleting` at some point, too which is better if we only prune a small amount of data.